src/java/org/apache/cassandra/net/OutboundConnections.java

```diff
@@ -95,7 +99,25 @@ static <K> OutboundConnections tryRegister(ConcurrentMap<K, OutboundConnections>
 
             if (existing == null)
             {
-                connections.metrics = new InternodeOutboundMetrics(settings.to, connections);
+                if (CUSTOM_INTERNODE_OUTBOUND_METRICS_PROVIDER_PROPERTY.isPresent())
```

Comment:
move to factory method in `InternodeOutboundMetrics` and make `InternodeOutboundMetrics` constructor inaccessible from other places.